siad/daemon.go

```diff
@@ -91,17 +82,16 @@ func startDaemon() error {
 	// that daemon startup has completed. A gofunc is used with the hope that
 	// srv.Serve() will start running before the value is sent down the
 	// channel.
+	//
+	// TODO: There are better ways to approach this.
```

Comment:
I like having the code coverage.

The main tests can't be very detailed but I like having them. I have some new ideas for how to approach them.